superset/db_engine_specs/hana.py

```diff
@@ -0,0 +1,59 @@
+from superset.db_engine_specs.base import LimitMethod
+from superset.db_engine_specs.postgres import PostgresBaseEngineSpec
+
+
+class HanaPyhdbEngineSpec(PostgresBaseEngineSpec):
+    engine = "hana+pyhdb"
+    limit_method = LimitMethod.WRAP_SQL
+    force_column_alias_quotes = True
+    max_column_name_length = 30
+    time_grain_functions = {
+        None: '{col}',
+        'P1D': "TO_DATE({col})",
+        'P1M': "SUBSTRING(to_date({col}),0,7)||'-01'",
+        'P0.25Y': "SUBSTRING(to_date({col}),0,5)||'0'||SUBSTRING(QUARTER(TO_DATE({col}), 1),7,1)||'-01'",
+        'P1Y': "YEAR({col})||'-01-01'",
```

Comment:
It seems `P1D` returns a `DATE` type while the others return some form of `STRING`. I believe these should all be cast to a `DATE`/`DATETIME` type or similar. Furthermore, would you be able to provide definitions for other time grains, like week, minute and second?

Reply:
Sorry, my English is poor. I have tested these well in my production environment and found no potential problems. However, I am not familiar with the standardized test, so I did not make the test report.Use multiple methods because many people do not install the standard sqlalchemy-hana and hdbcli libraries, please be aware that installing other libraries will also solve the hana link problem.
![image](https://user-images.githubusercontent.com/44701753/66691905-62685900-eccc-11e9-95c8-91062840903f.png)

Reply:
The hana background data stored in oracle is the character type (varchar), which I did not change to the time type, so it is the string type in this case.